src/lib/track.js

```diff
@@ -0,0 +1,52 @@
+/* @flow */
+import { getClientID, getMerchantID } from '@paypal/sdk-client/src';
+
+import { getUserIdCookie, setRandomUserIdCookie } from './cookie-utils';
+import { getDeviceInfo } from './get-device-info';
+import type {
+    Config,
+    TrackingType
+} from './types';
+
+export const track = <T>(config : Config, trackingType : TrackingType, trackingData : T) => {
+    const encodeData = data => encodeURIComponent(btoa(JSON.stringify(data)));
+
+    const img = document.createElement('img');
+    img.style.display = 'none';
+    if (!getUserIdCookie()) {
+        setRandomUserIdCookie();
+    }
+    const user = {
+        ...config.user,
+        id: getUserIdCookie()
+    };
+
+    // remove null, undefined values
+    user.id || delete user.id
+    user.email || delete user.email
+    user.name || delete user.name
```

Comment:
I thought setting `email` and `name` field to `null` or `''` will reset it on our back end? We want these values to get passed down the pipeline to get processed as is, so maybe we should not delete these fields?